.github/actions/docker-build/action.yaml

```
@@ -55,7 +55,10 @@ runs:
       shell: bash
       id: determine_tag_name
       run: |
-        if [ "$GITHUB_REF" = "refs/heads/main" ]; then
+        if [[ "$GITHUB_REF" =~ ^refs/tags/.* ]]; then
+          tag_name=`echo $GITHUB_REF | sed -e 's:^refs/tags/::'`
+          primary_tag="${tag_name}"
+        elif [ "$GITHUB_REF" = "refs/heads/main" ]; then
```

Probably it's possible to make it simpler by using GitHub actions context?

---

Which GitHub actions context are you referring to?

---

`github.ref` for example? Not fully understand what you are trying to achieve here. You can ignore it if you already have thought about it.
https://docs.github.com/en/actions/learn-github-actions/contexts

---

I didn't thought about it but seems GITHUB_REF is the same as github.ref. 

https://docs.github.com/en/actions/learn-github-actions/contexts#determining-when-to-use-contexts
```
Once the job is sent to the runner, the step is executed and refers to the [$GITHUB_REF](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables) variable from the runner.
```

I had a quick look on the document and seems there are no useful context we can use here.

This PR fix the missing logic to handle tag release on docker, leading the docker image "unknown" released.
https://hub.docker.com/layers/vdaas/vald-onnx-ingress-filter/unknown/images/sha256-aa9758ae8ed412e1b79716c8047cceed53203b837d8307087905bbc78706caf4?context=explore

I have referenced the code here:
https://github.com/vdaas/vald/blob/main/.github/actions/docker-build/action.yaml#L67-L87

---

> I have referenced the code here

Understood.

> I had a quick look on the document and seems there are no useful context we can use here.

I'm not sure about it though. It seems like `tag_name` can be `github.ref_name` for example. `github.event` can be used somehow I guess.

---

Yup, you are correct that `github.ref_name` can be used on `tag_name` here.
Something like this?
```suggestion
        if [[ "$GITHUB_REF" =~ ^refs/tags/.* ]]; then
          primary_tag="${{ github.ref_name }}"
        elif [ "$GITHUB_REF" = "refs/heads/main" ]; then
```

---

Yes. That's something I was thinking.

---

I will apply this changes  later

---

applied